core/amber/src/main/scala/edu/uci/ics/amber/engine/common/storage/partition/PartitionDocument.scala

```diff
@@ -0,0 +1,49 @@
+package edu.uci.ics.amber.engine.common.storage.partition
+
+import edu.uci.ics.amber.engine.common.storage.{FileDocument, VirtualDocument}
+
+import java.net.URI
+
+// PartitionDocument is a storage object that consists #numOfPartition physical files as its underlying data storage
+// Each underlying file's URI is in the format of {partitionDocumentURI}_{index}
+class PartitionDocument(val uri: URI, val numOfPartition: Int)
+    extends VirtualDocument[FileDocument] {
+  private def getPartitionURI(i: Int): URI = {
+    if (i < 0 || i >= numOfPartition) {
+      throw new RuntimeException(f"Index $i out of bound")
+    }
+
+    new URI(s"${uri}_$i")
+  }
+
+  override def getURI: URI =
+    throw new RuntimeException(
+      "Partition Document doesn't physically exist. It is invalid to acquire its URI"
+    )
+
+  override def readItem(i: Int): FileDocument = {
+    new FileDocument(getPartitionURI(i))
+  }
+
+  override def read(): Iterator[FileDocument] =
```

Comment:
usually, when you read something, the output is a copy of the original data, or the caller cannot use such a reference to remove things.  but here you can write `readItem(0).remove()` which is counter intuitive.

Reply:
I am thinking about changing these API's name:
- `read(): Iterator[T]` => `getIterator(): Iterator[T]`
- `readItem(i: Int): T` => `get(i: Int)`

I think this name is better as it also include the semantics that the returned item is modifiable.